### PR TITLE
Fix findings in Framework report

### LINKED ISSUE #8343: Fix findings found after the analysis of the Framework/Cluster code
|Wazuh version|Component|
|---|---|
| master | Wazuh Framework - Cluster |

Hi team,

As we have commented in issue #7998, there are several functions in the framework and cluster code where a random number is calculated and used to identify a certain task or log.

This can cause that at a certain moment, even if the probability is low, a random number will be calculated for a task and this number was previously assigned to another task. This would cause the loss of the first task.

On the other hand, in several parts of the Framework code we handle XML files. These files are not sanitized, so these parts of the code are vulnerable to possible attacks such as https://en.wikipedia.org/wiki/Billion_laughs_attack. To solve this problem we must use the `defusedxml` library. Introduced in the Framework requirements.

More information here: https://github.com/wazuh/wazuh/issues/7998#issuecomment-824792766

### ToDo

- [x] Refactor pseudo-random generators
- [x] Sanitize XML files

Regards,

Adrián Peña

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [ ] Affected tests 
  - [ ] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [ ] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

### PR DESCRIPTION
**Pull request by AdriiiPRodri**
|Related issue|
|---|
|#8343|

Hi team,

This PR closes #8343. In this PR we have fixed all the errors found by Bandit. In general we have fixed possible bugs. We managed to improve the overall stability of the Framework/Cluster.

Regards

### CODE REVIEW

File: framework/wazuh/core/utils.py [hunk 1]
```diff
@@ -1644,7 +1646,7 @@ def escape_formula_values(xml_string):
         return xml_string
 
     # Path of temporary files for parsing xml input
-    tmp_file_path = '{}/tmp/api_tmp_file_{}_{}.tmp'.format(common.wazuh_path, time.time(), random.randint(0, 1000))
+    tmp_file_path = '{}/tmp/api_tmp_file_{}_{}.tmp'.format(common.wazuh_path, time.time(), str(uuid4().hex[:8]))
```
Comment (davidjiglesias) [inline framework/wazuh/core/utils.py:1649]: I would use mkstemp

File: framework/wazuh/core/configuration.py [hunk 2]
```diff
@@ -658,7 +655,7 @@ def upload_group_configuration(group_id, file_content):
     if not os_path.exists(os_path.join(common.shared_path, group_id)):
         raise WazuhResourceNotFound(1710, group_id)
     # path of temporary files for parsing xml input
-    tmp_file_path = os_path.join(common.wazuh_path, "tmp", f"api_tmp_file_{time.time()}_{random.randint(0, 1000)}.xml")
+    tmp_file_path = os_path.join(common.wazuh_path, "tmp", f"api_tmp_file_{time.time()}_{str(uuid4().hex[:8])}.xml")
```
Comment (davidjiglesias) [inline framework/wazuh/core/configuration.py:658]: I would use mkstemp

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
Why use uuid4 instead of something like timestamp?

**Comment by AdriiiPRodri** [comment]
> Why use uuid4 instead of something like timestamp?

The reason for this is that if in the future the cluster/framework begins to use threads, although it is very unlikely, it could be the case that the same timestamp is generated in two or more threads, this will depend strongly on the accuracy of the timestamp. Because of this, using uuid4 seems easier, less expensive and with no probability of repeats.

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**